src/memory.cpp

```diff
@@ -341,7 +341,7 @@ bool cvk_image::init() {
 
     if (m_desc.image_type == CL_MEM_OBJECT_IMAGE1D_BUFFER) {
         auto buffer = static_cast<cvk_mem*>(m_desc.buffer);
-        m_memory = std::unique_ptr<cvk_memory_allocation>(buffer->memory());
+        m_memory = buffer->memory();
         buffer->retain();
     } else {
         // Select memory type
```

Comment:
You could have used `make_shared` in this case too. The code as it is is well-defined so not a problem, it's just a bit inconsistent.